### ISSUE TITLE
IllegalStateException

### ISSUE THREAD

**Issue by rmarau** [h6ah4i/android-advancedrecyclerview#247]
Hi

I'm hitting an IllegalStateException coming from the RecyclerViewDragDropManager.
Trace top:

> java.lang.IllegalStateException: Invalid range specified --- does not contain drag target item (range = ItemDraggableRange{mStart=0, mEnd=25}, position = -1)
>                                                        at com.h6ah4i.android.widget.advrecyclerview.draggable.RecyclerViewDragDropManager.verifyItemDraggableRange(RecyclerViewDragDropManager.java:879)

This occurs when dragging a group Item. But in _onCheckGroupCanStartDrag_ I'm closing all expanded items, as I don't want any expanded while dragging (similar to #200). 

This closing normally goes well. Except when I start dragging from outside the _drag_handle_ (yes it's possible) and move inside before the list scrolling catches the gesture. That is, starting to drag right on the outer edge of the _drag_handle_.

I've prepared a scenario in [branch](https://github.com/rmarau/android-advancedrecyclerview/tree/reproduce) that forces to close all expanded groups and reduces the size of the _drag_handle_ area.
Using Advanced > Expandable+Draggable+Swipeable:
 - expand any item
 -  then drag an item up (can be the same or not) but starting the drag from outside the _drag_handle_ area - just a few pixels down. Better to use a mouse point.
 - You'll get the IllegalStateException.

If you start from inside the _drag_handle_ it's all fine in closing the expanded groups.

I my case, I don't mind if the drag can start from an ACTION_DOWN event from outside, but this is leading to some unexpected state in the lib's manager.

Please, let me know if there's something unclear.

Thanks

**Comment by h6ah4i** [comment]
Hi. Thank you for the issue report. I have just fixed these two issues in the following branch.

- https://github.com/h6ah4i/android-advancedrecyclerview/compare/fix_issue_247

Could you try it through JitPack?

```gradle
repositories {
    maven { url "https://jitpack.io" }
}

dependencies {
    compile 'com.github.h6ah4i:android-advancedrecyclerview:39afacf660835c913bc8dd038378d502d60e019a'
}
```
---

> Except when I start dragging from outside the drag_handle (yes it's possible) and move inside before the list scrolling catches the gesture. That is, starting to drag right on the outer edge of the drag_handle.

I think this behavior should be fixed, so I added a `onCheckCanStartDrag()` method call on ACTION_DOWN touch event.

>  But in onCheckGroupCanStartDrag I'm closing all expanded items, as I don't want any expanded while dragging (similar to #200).

I assumed that `notifyXXX()` methods never be called during the `onCheckCan[Group|Child]StartDrag()` method. However it can be happen if library users call them explicitly or implicitly (`collapseAll()` for instance). I have added a verification after calling the `onCheckCanStartDrag()` method not to cause `IllegalStateException`.

Thanks.

**Comment by rmarau** [comment]
Hi
The fix solves the problem. Now the drag will start only if touching down inside the draggable area. And also not causing the `IllegalStateException`

Thanks for the follow up.

**Comment by h6ah4i** [comment]
@rmarau Hi. I have just released v0.9.2 and it includes the fix of this issue 😉

**Comment by rmarau** [comment]
Hi. It passed my test. Thanks again
